src/vs/workbench/services/environment/electron-sandbox/shellEnvironmentService.ts

```diff
@@ -0,0 +1,27 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { IProcessEnvironment } from 'vs/base/common/platform';
+import { process } from 'vs/base/parts/sandbox/electron-sandbox/globals';
+import { registerSingleton } from 'vs/platform/instantiation/common/extensions';
+
+export const IShellEnvironmentService = createDecorator<IShellEnvironmentService>('shellEnvironmentService');
+
+export interface IShellEnvironmentService {
+
+	readonly _serviceBrand: undefined;
+
+	readonly withShellEnv: Promise<IProcessEnvironment>;
```

Comment:
Let's drop the `when*` anti-pattern:

```suggestion
	getShellEnvironment(): Promise<IProcessEnvironment>;
```

Reply:
I went with `getShellEnv()` here